File: pypp_cli/test_dir/python/src/built_in_functions/list_fn.py
def built_in_list_fn():
    print("BUILT-IN LIST FUNCTION RESULTS:")
    a: list[int] = [1, 2, 3]
    b: list[int] = list(a)
    a.append(4)
    print(a)
    print(b)
    c: list[int] = list({1, 2, 3})
    print(c)
    d: list[str] = list("hello")
    print(d)
    e: dict[str, int] = {"one": 1, "two": 2}
    f: list[str] = list(e)
    print(f)
    g: list[str] = list[str]()
    print(g)

File: pypp_cli/test_dir/cpp/src/built_in_functions/list_fn.cpp
#include "built_in_functions/list_fn.h"
#include "py_dict.h"
#include "py_dict_default.h"
#include "py_list.h"
#include "py_set.h"
#include "py_str.h"
#include "pypp_util/create/list.h"
#include "pypp_util/print.h"

namespace me {
void built_in_list_fn() {
    pypp::print(pypp::PyStr("BUILT-IN LIST FUNCTION RESULTS:"));
    pypp::PyList<int> a({1, 2, 3});
    pypp::PyList<int> b = pypp::list(a);
    a.append(4);
    pypp::print(a);
    pypp::print(b);
    pypp::PyList<int> c = pypp::list(pypp::PySet({1, 2, 3}));
    pypp::print(c);
    pypp::PyList<pypp::PyStr> d = pypp::list(pypp::PyStr("hello"));
    pypp::print(d);
    pypp::PyDict<pypp::PyStr, int> e = {{pypp::PyStr("one"), 1},
                                        {pypp::PyStr("two"), 2}};
    pypp::PyList<pypp::PyStr> f = pypp::list(e);
    pypp::print(f);
    pypp::PyList<pypp::PyStr> g = pypp::PyList<pypp::PyStr>();
    pypp::print(g);
    auto h = pypp::PyDefaultDict<int, pypp::PyStr>::str_factory(
        {{1, pypp::PyStr("one")}});
    pypp::PyList<int> i = pypp::list(h);
    pypp::print(i);
}

} // namespace me